Sources/SwiftInspectorVisitors/Tests/StructVisitorSpec.swift

```diff
@@ -0,0 +1,274 @@
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+import Foundation
+import Nimble
+import Quick
+import SwiftInspectorTestHelpers
+
+@testable import SwiftInspectorVisitors
+
+final class StructVisitorSpec: QuickSpec {
+  private var sut = StructVisitor()
+
+  override func spec() {
+    beforeEach {
+      self.sut = StructVisitor()
+    }
+
+    describe("visit(_:)") {
+      context("visiting a single struct declaration") {
+        context("with no conformance") {
+          it("finds the type name") {
+            let content = """
+              public struct SomeStruct {}
+              """
+
+            try VisitorExecutor.walkVisitor(
+              self.sut,
+              overContent: content)
+
+            expect(self.sut.structs.first) == StructInfo(
+              name: "SomeStruct",
+              inheritsFromTypes: [],
+              parentTypeName: nil)
+          }
+        }
+
+        context("with a single type conformance") {
+          it("finds the type name") {
+            let content = """
+              public struct SomeStruct: Equatable {}
+              """
+
+            try VisitorExecutor.walkVisitor(
+              self.sut,
+              overContent: content)
+
+            expect(self.sut.structs.first) == StructInfo(
+              name: "SomeStruct",
+              inheritsFromTypes: ["Equatable"],
+              parentTypeName: nil)
+          }
+        }
+
+        context("with multiple type conformances") {
+          it("finds the type name") {
+            let content = """
+              public struct SomeStruct: Foo, Bar {}
+              """
+
+            try VisitorExecutor.walkVisitor(
+              self.sut,
+              overContent: content)
+
+            expect(self.sut.structs.first) == StructInfo(
+              name: "SomeStruct",
+              inheritsFromTypes: ["Foo", "Bar"],
+              parentTypeName: nil)
+          }
+        }
+      }
+
+      context("visiting a code block with nested declarations") {
+        context("visiting a struct with nested structs") {
+          it("finds all nested structs") {
+            let content = """
+              public struct FooStruct {
+                public struct BarFooStruct: Equatable {
+                  public struct BarBarFooStruct: Hashable {}
+                }
+                public struct FooFooStruct {
+                  public struct BarFooFoo1Struct: BarFooFoo1Protocol1,
+                    BarFooFoo1Protocol2
+                  {
+                    public struct BarBarFooFoo1Struct {}
+                  }
+                  public struct BarFooFoo2Struct {}
+                }
+              }
+              """
+
+            try VisitorExecutor.walkVisitor(
+              self.sut,
+              overContent: content)
+
+            expect(self.sut.structs) == [
```

Comment:
I'd recommend 2 things here

1) divide these into one it per struct
2) don't initialize the struct directly, but instead check their properties:

```swift
expect(someStruct.name) == "FooStruct"
expect(someStruct. inheritsFromTypes) == []
...
```

This way if we were to add 1 property to `StructInfo`, we don't need to modify n tests, we just modify the tests we care about.

Reply:
😬 I don't love the idea of having to access each item in the array individually. But I do like the idea of checking properties... so... okay fine 😉

Reply:
e2b5e143a2fc7568a76bf5fa11ba9f6a4db10dd8